server/lib/api/matchmakingTimes.ts

```diff
@@ -0,0 +1,173 @@
+import Koa from 'koa'
+import Router from '@koa/router'
+import httpErrors from 'http-errors'
+import Joi from 'joi'
+
+import { checkAllPermissions } from '../permissions/check-permissions'
+import ensureLoggedIn from '../session/ensure-logged-in'
+import { MATCHMAKING_TYPES } from '../../../common/constants'
+import { MATCHMAKING } from '../../../common/flags'
+import { AddMatchmakingTimeBody } from '../../../common/matchmaking'
+import { featureEnabled } from '../flags/feature-enabled'
+import { joiValidator } from '../validation/joi-validator'
+import {
+  getCurrentMatchmakingTime,
+  getFutureMatchmakingTimes,
+  getPastMatchmakingTimes,
+  addMatchmakingTime,
+  getMatchmakingTimeById,
+  removeMatchmakingTime,
+} from '../models/matchmaking-times'
+
+const matchmakingTypeSchema = Joi.object({
+  matchmakingType: Joi.valid(...MATCHMAKING_TYPES).required(),
+})
+
+const addMatchmakingTimeSchema = Joi.object({
```

Comment:
I think you might need to make this entire object required. You can check by trying a request that does not pass a request body

Reply:
Hmm, it does seem to work without that from what I can tell. Passing no body, I get this on the client:

![image](https://user-images.githubusercontent.com/4757944/102783415-2a15dd80-439b-11eb-9eab-c0154bb6244b.png)

Reply:
Cool, just wanted to make sure since I saw some weirdness but it was with requests made from Rust, and I chose to err on the safer side rather than recompiling/restarting games a bunch of times to find out :)